sympy/solvers/ode.py

```diff
@@ -4003,6 +4012,42 @@ def _frobenius(n, m, p0, q0, p, q, x0, x, c, check=None):
 
     return frobdict
 
+def _check_substitution_type_match(eq, func):
```

Comment:
suggest:
```python
def _check_substitution_type_match(eq, func):
    r"""
    Matches any differential equation that `order_reducing_substitution` can solve.
    For this to work equation should have a minimum order of derivative to be 1, i.e.,
    `f(x)` should not be present.
    """
    fx = func
    assert len(func.args) == 1  # ODE only handles functions of 1 variable
    x = func.args[0]
    D = Dummy()
    for d in ordered(eq.atoms(Derivative)):
        try:
            (v, c), = d.variable_count  # ValueError if not a singleton
            if v != x:
                raise ValueError  # not a derivative of interest
        except ValueError:
            continue
        # d was the lowest-ordered derivative of interest
        if eq.subs(d, D).has(fx):
            return  # it didn't clear all occurrences of f(x)
        return {'var': c}
```

Can be tested with
```
w, x, y= symbols('w x y')
f = Function('f')
F = lambda eq: _check_substitution_type_match(eq, f(x))
fx = f(x)
D = Derivative
assert F(D(y*fx, x, y) + D(fx, x)) is None
assert F(D(y*f(y), y, y) + D(f(y), y)) is None
assert F(fx*D(fx, x) + D(fx, x, 2)) is None
assert F(D(x*f(y), y, 2) + D(w*y*fx, x, 3)) == dict(var=3)
assert F(D(f(w), w, 2) + D(f(w), w, 3) + D(fx, x, 4)) == dict(var=4)
```

If you don't like the suggestion, please put space after comma, return only `{'var': order_to_subs}`.